packages/core/src/customer/strategies/stripe-upe/stripe-upe-customer-strategy.ts

```diff
@@ -80,14 +80,24 @@ export default class StripeUPECustomerStrategy implements CustomerStrategy {
                 appearance,
             });
 
-            const linkAuthenticationElement = this._stripeElements.getElement(StripeElementType.AUTHENTICATION) || this._stripeElements.create(StripeElementType.AUTHENTICATION);
+            const billingAddress = this._store.getState().billingAddress.getBillingAddress();
+            const options = billingAddress?.email? { defaultValues: { email: billingAddress?.email } } : { } ;
+            const linkAuthenticationElement = this._stripeElements.getElement(StripeElementType.AUTHENTICATION) ||
+                this._stripeElements.create(StripeElementType.AUTHENTICATION, options);
 
             linkAuthenticationElement.on('change', (event: StripeEventType) => {
                 if (!('authenticated' in event)) {
                     throw new MissingDataError(MissingDataErrorType.MissingCustomer);
                 }
-                this._store.dispatch(createAction(CustomerActionType.StripeLinkAuthenticated, event.authenticated));
-                event.complete ? onEmailChange(event.authenticated, event.value.email) : onEmailChange(false, '');
+                if (event.complete) {
+                    onEmailChange(event.authenticated, event.value.email);
+                    if (isStripeLinkAuthenticated === undefined && sessionStorage.getItem('stripeLink') != StripeSessionStorageKey.CUSTOMER_RELOADED) {
```

Comment:
```suggestion
                    if (isStripeLinkAuthenticated === undefined && sessionStorage.getItem('stripeLink') !== StripeSessionStorageKey.CUSTOMER_RELOADED) {
```

Reply:
I am still confused as to why we need to set/get the value from session storage? Since from what I remember on reload the shopper will come to customer step where we have the email from billing address and that should be used to validate stripe link?

Reply:
@animesh1987, basically, the sessionStorage approach is used to cover these two fundamental needs in a non-aggressive way:

1 be able to identify that stripeLink is being used

2 we want to be able to do a kind of auto-step in the stripeLink components only the first time of the reload, that's why the change of value in the variable